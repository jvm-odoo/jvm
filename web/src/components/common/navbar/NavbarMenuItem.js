import styled from 'styled-components'

export const NavbarMenuItem = styled.li`
  display: inline-block;
  height: 100%;
  width: 175px;

  a {
    display: flex;
    justify-content: center;
    align-items: center;
    color: ${({ theme }) => theme.title.default};
    text-decoration: none;
    font-weight: 600;
    transition: 0.2s;
    height: 100%;

    :visited {
      color: ${({ theme }) => theme.title.default} !important;
    }
  }

  :hover {
    a {
      transform: translateY(-2px);
    }
  }
`
