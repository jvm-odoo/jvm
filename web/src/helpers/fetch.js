export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
