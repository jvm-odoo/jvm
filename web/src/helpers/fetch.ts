export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}