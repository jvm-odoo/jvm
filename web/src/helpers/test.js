import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components'

import { store } from '../redux/store'
import { THEME_LIGHT } from '../constants'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(
    <ThemeProvider theme={THEME_LIGHT}>
      {Component}
    </ThemeProvider>)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ThemeProvider theme={THEME_LIGHT}>
          {Component}
        </ThemeProvider>
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}

export const withReduxProvider = (Component) => (
  <Provider store={store}>
    {Component}
  </Provider>
)
