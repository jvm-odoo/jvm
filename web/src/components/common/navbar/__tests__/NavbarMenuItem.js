import React from 'react'

import { NavbarMenuItem } from '../NavbarMenuItem'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<NavbarMenuItem />))
