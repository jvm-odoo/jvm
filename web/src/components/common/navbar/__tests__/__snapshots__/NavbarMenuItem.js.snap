// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "background": "#FEFEFE",
      "headerBorder": "#EEE",
      "mode": "light",
      "text": "#797979",
      "themeSwitcherShadow": "0px 0px 25px 0px rgba(0,0,0,0.15)",
      "title": Object {
        "default": "#24292E",
        "inverse": "#FEFEFE",
      },
      "wavyContainer": "linear-gradient(135deg, rgba(90,81,250,0.5) 0%, rgba(179,204,210,0.5) 100%)",
    }
  }
>
  <styled.li />
</ContextProvider>
`;
